csharp/src/Apache.Arrow.Adbc/Tracing/TracingReader.cs

```diff
@@ -47,7 +47,7 @@ protected virtual void Dispose(bool disposing)
         {
         }
 
-        public void Dispose()
+        public virtual void Dispose()
```

Comment:
This shouldn't be made `virtual` because the overload above should be overridden instead (as `DatabricksCompositeReader` is doing).